server/src/main/java/org/elasticsearch/plugins/PluginsLoader.java

```diff
@@ -370,6 +384,16 @@ static String toModuleName(String name) {
         return result;
     }
 
+    static String toModuleName2(String name) {
```

Comment:
Very hacky, but I did not want to break the existing name schema for stable plugins. 
And there is a `ml-package-loader` loader unfortunately -- `ml.package.loader` is not a OK module name, as `package` is a reserved keyword.